lib/flow/account.ts

```diff
@@ -1,31 +1,41 @@
 import * as fcl from "@onflow/fcl"
 import * as t from "@onflow/types"
-import {encodeKey} from "@onflow/util-encode-key"
+import { encodeKey } from "@onflow/util-encode-key"
 import publicConfig from "../publicConfig"
-import {sendTransaction} from "./send"
+import { sendTransaction } from "./send"
 
 const accountCreatedEventType = "flow.AccountCreated"
 
 const txCreateAccount = `
 import FlowToken from ${publicConfig.contractFlowToken}
 import FungibleToken from ${publicConfig.contractFungibleToken}
 
-transaction(publicKey: String, flowTokenAmount: UFix64) {
+transaction(publicKey: String, flowTokenAmount: UFix64, sigAlgorithm: UInt8, hashAlgorithm: UInt8) {
   let tokenAdmin: &FlowToken.Administrator
   let tokenReceiver: &{FungibleToken.Receiver}
 
-  prepare(signer: AuthAccount) {
-    let account = AuthAccount(payer: signer)
+  prepare(signer: auth(BorrowValue) &Account) {
+    let account = Account(payer: signer)
 
-    account.addPublicKey(publicKey.decodeHex())
```

Comment:
This method was deprecated, so `sigAlgorithm` & `hashAlgorithm` were added as transaction parameters to better match the preceding call to `encodeKey` which took `number` values to define each algo. This allows `createAccount` interface (defined below) to remain the same.